hack/v1migration/phase1.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+set -e
+
+if [ $# != 1 ]; then
+	echo "USAGE: $0 [WORKDIR]"
+	exit 1
+fi
+
+WORKDIR=$1
+echo "Using workdir: $WORKDIR"
+
+oc project hive
+
+oc scale -n hive deployment.v1.apps/hive-operator --replicas=0
+oc scale -n hive deployment.v1.apps/hive-controllers --replicas=0
+oc scale -n hive deployment.v1.apps/hiveadmission --replicas=0
+
+# TODO: wait until no pods are running
+
+mkdir -p $WORKDIR/
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+for i in "${HIVE_TYPES[@]}"
+do
+	:
+	echo "Storing all ${i} in ${WORKDIR}/${i}.yaml"
+	oc get ${i}.hive.openshift.io -A -o json | jq '.items | .[] |
+		del(.status) |
+		del(.metadata.annotations."kubectl.kubernetes.io/last-applied-configuration") |
+		del(.metadata.creationTimestamp) |
```

Comment:
Do we really need to strip this kind of stuff out? I would expect the API server to strip it out for us when we go to create the objects. Or is the purpose of stripping it out to reduce the size of the yaml?

Reply:
These examples are partly based on our scripts for duplicating clusters, I actually thought the apiserver would reject a create with a creationtimestamp, uid, etc. I'd have to test though.

Reply:
Ah, you may be right.